python/_fio_plugin/fio_plugin.py

```diff
@@ -0,0 +1,52 @@
+import sys
+import typing
+import json
+import subprocess
+import dataclasses
+from traceback import format_exc
+from typing import Union
+
+from arcaflow_plugin_sdk import plugin
+from fio_schema import FioParams, FioSuccessOutput, FioErrorOutput, fio_output_schema
+
+
+@plugin.step(
+    id="workload",
+    name="fio workload",
+    description="run an fio workload",
+    outputs={"success": FioSuccessOutput, "error": FioErrorOutput}
+)
+def run(params: FioParams) -> typing.Tuple[str, Union[FioSuccessOutput, FioErrorOutput]]:
+    try:
+        outfile_name = 'fio-plus'
+        cmd = [
+            'fio',
+            *[
+                f"--{key}={str(value)}" for key, value in dataclasses.asdict(params).items()
```

Comment:
I always get a little worried when I see us passing command line flags in a loop. There are a few ways this can get problematic. Is there a good reason here to use the command flag approach instead of the fio supported job file?